driver-core/build.gradle

```diff
@@ -47,6 +47,12 @@ dependencies {
     api "io.netty:netty-buffer", optional
     api "io.netty:netty-transport", optional
     api "io.netty:netty-handler", optional
+
+    // Optionally depend on both AWS SDK v2 and v1.  The driver will use v2 is present, v1 if present, or built-in functionality if
+    // neither are present
+    implementation "software.amazon.awssdk:auth:$awsSdkV2Version", optional
+    implementation "com.amazonaws:aws-java-sdk-core:$awsSdkV1Version", optional
```

Comment:
Would it be worth making v1 a compileOnly dependency and add it to testImplementation.  That way it wouldn't show up in the pom but still can be found if on the runtime path? It would be another hint to use the v2 version.

Reply:
I'm not sure... According to the [blog](https://blog.gradle.org/introducing-compile-only-dependencies) introducing compileOnly dependencies:

> Dependencies required at compile time but required at runtime only when using certain features, a.k.a. optional dependencies

It sounds applicable to _any_ optional dependency, which would include both v1 and v2 at this point, as well as a few others that we have.  Should they all be compileOnly? It's not clear to me when to use compileOnly and when to use optional (is optional provided natively, or via plugin?).

Reply:
Just saw this in the release notes for compileOnly:

> However, unlike Maven provided dependencies, compile only dependencies in Gradle are not included on the test classpath

But we need this in our test classpath at least for some tasks.

Reply:
So m,y question was really about about signalling usage in the pom.xml.  If we add it as a `compileOnly` and `testImplementation` dependency, it won't be mentioned in the pom.xml but will still be testable.

So do we want to advertize it as an optional dependency as well as v2 or just make it an implicit but tested dependency.